src/shogun/lib/Signal.h

```diff
@@ -30,7 +30,7 @@ namespace shogun
 	 *  option bewteen: immediately exit the running method and fall back to
 	 *  the command line, prematurely stop the current algoritmh and do nothing.
 	 */
-	class CSignal : public CSGObject
+	class CSignal
```

Comment:
doesn't need to inherit from SGObject + it caused problems (ShogunEnv tries to create CSignal, and CSignal tries to create ShogunEnv)